src/snowflake/connector/arrow_context.py

```diff
@@ -148,3 +149,11 @@ def TIMESTAMP_NTZ_TWO_FIELD_to_numpy_datetime64(
     ) -> datetime64:
         nanoseconds = int(decimal.Decimal(epoch).scaleb(9) + decimal.Decimal(fraction))
         return numpy.datetime64(nanoseconds, "ns")
+
+    def DECIMAL128_to_decimal(self, int128_bytes: bytes, scale: int):
+        int128 = int.from_bytes(int128_bytes, byteorder=byteorder, signed=True)
+        if scale == 0:
+            return int128
+        digits = [int(digit) for digit in str(int128) if digit != "-"]
+        sign = int128 < 0
+        return decimal.Decimal((sign, digits, -scale))
```

Comment:
Dumb question: why is this `-scale`?

Reply:
I will let the code talk, this is the python behavior to control the digits allowed to the right of the decimal point:

```python
ret = decimal.Decimal((True, [1,2,3,4,5], -3))
# ret is 12.345
ret = decimal.Decimal((True, [1,2,3,4,5], -2))
# ret is 123.45
```